packages/styled-components/src/sheet/Rehydration.ts

```diff
@@ -16,7 +16,7 @@ export const outputSheet = (sheet: Sheet) => {
 
     const names = sheet.names.get(id);
     const rules = tag.getGroup(group);
-    if (names === undefined || rules.length === 0) continue;
+    if (names === undefined || !names.size || rules.length === 0) continue;
```

Comment:
I'm not sure about this change w.r.t. createGlobalStyle. Could you add a test verifying that cGS usage without any other components works as expected in SSR?

Reply:
Test added